### PR TITLE
feat: added github actions

### PR DESCRIPTION
**Pull request by Abdul-SubhanL**

### CODE REVIEW

**Review by sumairruhani2 — CHANGES_REQUESTED** [review]
You committed both a package-lock.json and yarn.lock file. You only need to have one or the other as they both do the same thing (just one uses yarn and the other uses npm for package management)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Set up Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run tests
+        run: npm test
```
Comment (sumairruhani2) [inline .github/workflows/test.yml]: This wont work as we don't have the jest dependency

**Review by sumairruhani2 — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (Abdul-SubhanL) [reply .github/workflows/test.yml]: Oh good spot, added Jest dependency 👍

Comment (Abdul-SubhanL) [reply .github/workflows/test.yml]: It will fail until tests get added

Comment (Abdul-SubhanL) [reply .github/workflows/test.yml]: tests added

**Pull request merged**